### DISCUSSION TITLE
Making statically linked Windows application to be FIPS compliant while dependencies are build using vcpkg

### DISCUSSION THREAD

**Discussion by okravets** [openssl/openssl#23644] [Q&A] [answered]
Hey,

I need to make our application to be FIPS compliant. 

According to documentation, I need to use 3.0.8 FIPS compliant dll and install it on every machine.

Currently, our application is running as service on Windows and it is statically linked to all third-party libraries including OpenSSL.
Dependencies are built through vcpkg using x64-static configuration.

After reading documentation, it seems to me that I cannot rely on static linkage anymore. Also I see a lot of lines with FIPS_MODULE define.

I need clarification, if I still could compile and link OpenSSL statically with all dependencies using it and then in runtime load fips.dll?
Or I need to use dynamic linking for OpenSSL and ensure that all dependencies rely on OpenSSL dll?

Additionally I need to assure that none of deprecated API used by dependend libraries. Probably best bet, to set no-deprecated option for that.

Please advise.

Thank you,
Olexandr

**Comment by t8m** [comment, marked as answer]
As a first step I would recommend to read https://github.com/openssl/openssl/blob/master/README-FIPS.md

With current versions (3.0.13/3.1.5/3.2.1) you can actually build openssl with no-shared and link it statically to your application but still have a separate fips.dll provider module built from the validated sources (3.0.9 is the latest validated version).
